app/org/sagebionetworks/bridge/hibernate/AccountPersistenceExceptionConverter.java

```diff
@@ -46,8 +60,20 @@ public RuntimeException convert(PersistenceException exception, Object entity) {
                 // "Duplicate entry 'api-ext' for key 'Accounts-StudyId-ExternalId-Index'" 
                 EntityAlreadyExistsException eae = null;
                 if (message.matches("Duplicate entry.*for key 'Accounts-StudyId-ExternalId-Index'")) {
-                    eae = createEntityAlreadyExistsException("External ID",
-                            AccountId.forExternalId(account.getStudyId(), account.getExternalId()));
+                    // We do not know which external ID is the conflict without parsing the error message. 
+                    // Try them until we find one. This external ID could be in a sub-study the caller is 
+                    // not associated to, but this is because external IDs have to be unique at the scope 
+                    // of the study, in order to identify substudy membership FROM the external ID.
+                    account = (HibernateAccount)BridgeUtils.filterForSubstudy(account);
+                    if (account != null) {
+                        for (String externalId : BridgeUtils.collectExternalIds(account)) {
+                            eae = createEntityAlreadyExistsException("External ID",
+                                    AccountId.forExternalId(account.getStudyId(), externalId));
+                            if (eae != null) {
+                                break;
+                            }
+                        }
+                    }
```

Comment:
I have moved the filtering of subsidies out of the AccountDao because of strange errors that occurred when a user in one substudy triggered the loading of an account that was in another substudy. So no, in this handler all the substudies are there. Filtering now happens in the ParticipantService where it's needed because a record is being returned from the API.